internal/project/auto/golang.go

```diff
@@ -22,6 +23,10 @@ import (
 
 // DetectGolang checks if project at rootPath is Go-based project.
 func (builder *builder) DetectGolang() (bool, error) {
+	if builder.meta.ContainerImageFrontend != config.ContainerImageFrontendDockerfile {
```

Comment:
we can short circuit early, also helps adding conditional checks in so many other places